.github/workflows/tests.yml

```diff
@@ -280,10 +310,12 @@ jobs:
             test-report.xml
           retention-days: 1
 
-  analyze:
-    name: Analyze results
-    needs: [windows, linux, linux-qemu, macos]
-    if: always()
+  # aggregate and upload
+  aggregate:
+    # only aggregate test suite if there are code changes
+    needs: [changes, windows, linux, linux-qemu, macos]
+    if: needs.changes.outputs.code == 'true' && always()
```

Comment:
That reads a little weird, do we need the `always()` here?

Reply:
Apparently, we do. The `always()` function does more than just evaluate to `true`. I had previously just used `if: needs.changes.outputs.code == 'true'` but that resulted in this job being skipped if any of the `needs` failed.